BraveWallet/Crypto/NetworkSelectionView.swift

```diff
@@ -0,0 +1,336 @@
+      } else {
+        Image(systemName: "checkmark").hidden()
+      }
+    }
+    .foregroundColor(Color(.braveLabel))
+  }
+  
+  private var showShortChainName: Bool {
+    presentation.isPrimaryNetwork && !presentation.subNetworks.isEmpty
+  }
+
+  var body: some View {
+    HStack {
+      HStack {
+        checkmark
+        NetworkIcon(network: presentation.network)
+        VStack(alignment: .leading, spacing: 0) {
+          Text(showShortChainName ? presentation.network.shortChainName : presentation.network.chainName)
+          if presentation.subNetworks.contains(selectedNetwork) {
+            Text(selectedNetwork.chainName)
+              .foregroundColor(Color(.secondaryBraveLabel))
+              .font(.footnote)
+          }
+        }
+        Spacer()
+      }
+      .accessibilityElement(children: .combine)
+      .accessibilityAddTraits(isSelected ? [.isSelected] : [])
+      if !presentation.subNetworks.isEmpty {
+        Button(action: { detailTappedHandler?() }) {
+          Image(systemName: "chevron.right.circle")
+            .foregroundColor(Color(.braveBlurpleTint))
+            .contentShape(Rectangle())
+        }
+        .buttonStyle(.plain)
+        .accessibilityLabel(
+          Text(String.localizedStringWithFormat(
+            Strings.Wallet.networkSelectionTestnetAccessibilityLabel,
+            presentation.network.shortChainName)
+          )
+        )
+      }
+    }
+    .foregroundColor(Color(.braveLabel))
+    .padding(.vertical, 4)
+    .frame(maxWidth: .infinity, alignment: .leading)
+    .contentShape(Rectangle())
+  }
+}
+
+#if DEBUG
+struct NetworkRowView_Previews: PreviewProvider {
+  static var previews: some View {
+    Group {
+      NetworkRowView(
+        presentation: .init(
+          network: .mockSolana,
+          subNetworks: [.mockSolana],
+          isPrimaryNetwork: true
+        ),
+        selectedNetwork: .mockSolana
+      )
+      NetworkRowView(
+        presentation: .init(
+          network: .mockMainnet,
+          subNetworks: [.mockMainnet, .mockRinkeby, .mockRopsten],
+          isPrimaryNetwork: true
+        ),
+        selectedNetwork: .mockMainnet
+      )
+      NetworkRowView(
+        presentation: .init(
+          network: .mockPolygon,
+          subNetworks: [],
+          isPrimaryNetwork: false
+        ),
+        selectedNetwork: .mockMainnet
+      )
+    }
+    .previewLayout(.sizeThatFits)
+  }
+}
+#endif
+
+// MARK: Detail View
+
+private struct NetworkSelectionDetailView: View {
+  
+  var networks: [BraveWallet.NetworkInfo]
+  var selectedNetwork: BraveWallet.NetworkInfo
+  var selectedNetworkHandler: (BraveWallet.NetworkInfo) -> Void
+  
+  var body: some View {
+    List {
+      ForEach(networks) { network in
+        NetworkSelectionDetailRow(
+          isSelected: selectedNetwork == network,
+          network: network
+        )
+        .contentShape(Rectangle())
+        .onTapGesture {
+          selectedNetworkHandler(network)
```

Comment:
Can definitely use a Button in this case (no detail disclosure)

Reply:
Fixed in [c29f314](https://github.com/brave/brave-ios/pull/5706/commits/c29f314f45ae50559626c245a8c40fe2dbdfd34e)